packages/microsoft-security-adapter/src/definitions/references/references.ts

```diff
@@ -8,18 +8,42 @@
 import { definitions, references as referenceUtils, createChangeElementResolver } from '@salto-io/adapter-components'
 import { Change, InstanceElement } from '@salto-io/adapter-api'
 import { ODATA_TYPE_FIELD_NACL_CASE, entraConstants } from '../../constants'
-import { Options } from '../types'
-import { REFERENCE_RULES } from './entra_reference_rules'
+import { CustomReferenceSerializationStrategyName, Options } from '../types'
+import { REFERENCE_RULES as EntraReferenceRules } from './entra_reference_rules'
+import { REFERENCE_RULES as IntuneReferenceRules } from './intune_reference_rules'
+
+const REFERENCE_RULES = [...EntraReferenceRules, ...IntuneReferenceRules]
+
+// We only use this record as a TS 'hack' to fail the build if we add a custom serialization strategy
+// and forget to add it to the fieldsToGroupBy.
+const fieldsToGroupBy: Record<'id' | CustomReferenceSerializationStrategyName, null> = {
```

Comment:
I think this should be 
```suggestion
const fieldsToGroupBy: Record<ReferenceIndexField | CustomReferenceSerializationStrategyName, null> = {
```

this way if we add something to adapter-components this map will have to be updated as well